src/main/java/me/itzg/helpers/modrinth/ModrinthApiPackFetcher.java

```diff
@@ -0,0 +1,82 @@
+package me.itzg.helpers.modrinth;
+
+import java.nio.file.Path;
+
+import lombok.extern.slf4j.Slf4j;
+import me.itzg.helpers.errors.*;
+import me.itzg.helpers.files.Manifests;
+import me.itzg.helpers.http.FailedRequestException;
+import me.itzg.helpers.modrinth.model.*;
+import reactor.core.publisher.Mono;
+
+@Slf4j
+public class ModrinthApiPackFetcher implements ModrinthPackFetcher {
+    private ModrinthApiClient apiClient;
+
+    private ProjectRef modpackProjectRef;
+    private Mono<Version> modpackVersionStream;
+
+    private Loader modLoaderType;
+    private String gameVersion;
+    private VersionType defaultVersionType;
+    private Path modpackOutputDirectory;
+
+    ModrinthApiPackFetcher(InstallModrinthModpackCommand config) {
```

Comment:
Please avoid passing around the "command" object. Think of it like a REST controller/resource that is the glue between the outside caller and the inner code. Instead, break out the individual parameters and pass those to this constructor.